girder_annotation/girder_large_image_annotation/web_client/views/annotationListWidget.js

```diff
@@ -182,24 +193,104 @@ const AnnotationListWidget = View.extend({
         this.$el.find('th.g-annotation-toggle i').toggleClass('icon-eye', !anyOn).toggleClass('icon-eye-off', !!anyOn);
     },
 
-    _deleteAnnotation(evt) {
+    _selectAnnotation(evt) {
+        // Prevent event from bubbling up to the row click handler
+        // that toggles the annotation on and off
+        evt.stopPropagation();
+
         const $el = $(evt.currentTarget);
         const id = $el.parents('.g-annotation-row').data('annotationId');
+        const allChecks = this.$el.find('.g-annotation-select input[type=checkbox]');
+        const anySelected = this.$('.g-annotation-select input:checked').length > 0;
+
+        this.$('thead .g-annotation-download-selected, thead .g-annotation-delete, thead .g-annotation-permissions')
+            .prop('disabled', !anySelected)
+            .toggleClass('disabled', !anySelected)
+            .css('color', anySelected ? '' : 'grey');
         if (!id) {
-            confirm({
-                text: `Are you sure you want to delete <b>ALL</b> annotations?`,
-                escapedHtml: true,
-                yesText: 'Delete',
-                confirmCallback: () => {
-                    restRequest({
-                        url: `annotation/item/${this.model.id}`,
-                        method: 'DELETE'
-                    }).done(() => {
-                        this.collection.fetch(null, true);
+            if ($el.find('input').is(':checked')) {
+                allChecks.prop('checked', true);
+            }
+            if (!$el.find('input').is(':checked')) {
+                allChecks.prop('checked', false);
+            }
+        }
```

Comment:
After checking the check-all box, if you uncheck another box, the check-all box is still checked.  I thin we want an else to the `if (!id)` that would change the check-all box to be checked or unchecked based on if all of the other boxes are checked.

Reply:
updated in 256499718c11e0d6c10c437ab357c77d7d67e8ff